segment-kube.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+type KubeContext struct {
+	Context struct {
+		Cluster   string
+		Namespace string
+		User      string
+	}
+	Name string
+}
+
+type KubeConfig struct {
+	Contexts       []KubeContext `yaml:"contexts"`
+	CurrentContext string        `yaml:"current-context"`
+}
+
+func segmentKube(p *powerline) {
+	home, _ := os.LookupEnv("HOME")
+	defaultConfigFile := home + "/.kube/config"
+	kubeConfigFile := defaultConfigFile
+	kubeEnv, ok := os.LookupEnv("KUBECONFIG")
+
+	if ok {
+		possibleConfigs := strings.Split(kubeEnv, ":")
+		// for now just take the last one
+		// TODO: find one that works
+		kubeConfigFile = possibleConfigs[0]
+		if len(possibleConfigs) > 1 {
+			kubeConfigFile = possibleConfigs[len(possibleConfigs)-1]
+		}
+	}
+
+	//	fmt.Println("Reading config file " + kubeConfigFile)
+	path, err := filepath.Abs(kubeConfigFile)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	// parse the config
+	config, err := ioutil.ReadFile(path)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	kc := &KubeConfig{}
+	err = yaml.Unmarshal(config, kc)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	cluster := ""
+	namespace := ""
+	for _, c := range kc.Contexts {
+		if c.Name == kc.CurrentContext {
+			cluster = c.Context.Cluster
+			namespace = c.Context.Namespace
+			break
+		}
+	}
+
+	if strings.HasPrefix(cluster, "gke") {
```

Comment:
This should be commented to indicate we're truncating google container engine (GKE) cluster names from long-form to a shorter more human friendly form, eg:

```
gke_projectname_availability-zone_cluster-01 => cluster-01
```

Reply:
Maybe we should make this a config option?

Reply:
+1